### PR TITLE
Debian base image migration (rebased)

### PR DESCRIPTION
**Pull request by charles-boyd**
Opening a new PR to introduce the same changes as from #196 with a clean merge history.

### CODE REVIEW

File: securethenews/requirements.in [hunk 1]
```diff
@@ -6,7 +6,6 @@ django-filter>=2.1.0
 -e git+https://github.com/freedomofpress/django-logging.git@51f4fd4b02fafc3d6d9cddb5658247497404df7e#egg=django-logging-json
 django-mailgun
 django-storages[boto3, google]>=1.7 # aws s3, google storage asset management
-django-webpack-loader
```
Comment (conorsch) [inline securethenews/requirements.in]: We do indeed want `django-webpack-loader`! That's used to collect the static assets, as of #181.

File: docker/ProdDjangoDockerfile [hunk 2]
```diff
@@ -4,39 +4,41 @@ FROM node:${NODE_VER} AS node-assets
...
-ENV UV_THREADPOOL_SIZE 128
-
 # Oddly, node-sass requires both python and make to build bindings
 RUN apk add --no-cache paxctl python make g++
 RUN paxctl -cm /usr/local/bin/node
 
-COPY ./ /src-files
-RUN cd /src-files && ( npm install && npm run build )
+# Install gulp globally so it can be used during builds
+RUN NPM_CONFIG_UNSAFE_PERM=true npm install --global gulp-cli
```
Comment (conorsch) [inline docker/ProdDjangoDockerfile]: The re-inclusion of `gulp` is a regression: remove the install task for gulp, and reinstate the `npm install && npm run build` task above. See here for reference: https://github.com/freedomofpress/securethenews/pull/181/files#diff-39520f27c7fd6ecc9b11465fde9fff66R12

File: docker/ProdDjangoDockerfile [hunk 3]
```diff
@@ -4,39 +4,41 @@ FROM node:${NODE_VER} AS node-assets
 # Make npm output less verbose
 ENV NPM_CONFIG_LOGLEVEL warn
 
-# Workaround to avoid webpack hanging, see:
-# https://github.com/webpack/webpack-dev-server/issues/128
-ENV UV_THREADPOOL_SIZE 128
```
Comment (conorsch) [inline docker/ProdDjangoDockerfile]: If you've tested and the UV_THREADPOOL_SIZE override is not necessary for full builds, great! See comments in https://github.com/freedomofpress/securethenews/pull/198/commits/a0cddeb2a99e4dbc77d60919cb51f188def87cd5 ; I suspect it's worth preserving.

File: securethenews/requirements.in (hunk 1 above)
Comment (charles-boyd) [reply securethenews/requirements.in]: Not sure why this got removed. I have added it back in my working branch.

File: docker/ProdDjangoDockerfile (hunk 3 above)
Comment (charles-boyd) [reply docker/ProdDjangoDockerfile]: I have added this back in my working branch. Removal was a regression.

**Comment by conorsch** [comment]
Changes look great, @charles-boyd! I squashed your final fixes into the other commits, so we have a clean history to consult going forward. Provided CI passes, fine for merge. Will circle back on those infra tests shortly, once this lands.

**Pull request merged**